src/main/java/com/amazonaws/services/kinesis/leases/interfaces/ILeaseSerializer.java

```diff
@@ -74,6 +74,14 @@
      */
     public Map<String, ExpectedAttributeValue> getDynamoLeaseOwnerExpectation(T lease);
 
+    /**
+     * @param lease
+     * @return the attribute value map asserting that the checkpoint state is as expected.
+     */
+    default Map<String, ExpectedAttributeValue> getDynamoLeaseCheckpointExpectation(T lease) {
```

Comment:
Maybe good to rename this to something like `getDynamoLeaseShardEndCheckpointExpectation`?

Reply:
I just want to keep this generic, so that we can add any future state expectations. Also this would set expectation based on checkpoint in the lease.